src/js/net.js

```diff
@@ -403,15 +403,20 @@ function onread(socket, nread, isEOF, buffer) {
       return;
     }
 
-    var str = buffer.toString();
+    // We know for sure the last 6 characters are going to be the ending
+    // Lets create a buffer with those 6 characters without toString conversion.
+    var eofLength = 6;
+    var bufferLength = buffer.length;
+    var ending = new Buffer(eofLength);
+    buffer.copy(ending, 0, bufferLength - eofLength, bufferLength);
+    var expectedEnding = new Buffer('\\e\\n\\d');
```

Comment:
Do we really need to create this buffer every time the `onread` is triggered?

Reply:
I moved it to a global value, good point